game/domain/body.lua

```diff
@@ -397,22 +397,18 @@ function Body:removeAllArmor()
 end
 
 function Body:takeDamageFrom(amount, source)
-  local blocked = math.min(amount, self.armor)
-  self.armor = self.armor - blocked -- should never go negative
-  local dmg = math.max(0, amount - blocked)
+  local blocked = false
+  if RANDOM.generate(100) <= self:getBlockChance() then
+    amount = math.floor(amount/2)
+    blocked = true
+  end
+  local absorbed = math.min(amount, self.armor)
+  self.armor = self.armor - absorbed -- should never go negative
```

Comment:
This comment begs for unit testing

Reply:
(if you're going to add testing, don't do it in this PR, do it in another as it's already big enough of a PR)

Reply:
Yes, this line was actually already in the code, so it's really unrelated.

Reply:
Created issue #1001 for this